src/scenes/NftDetailPage/NftAdditionalDetails.tsx

```diff
@@ -12,8 +12,15 @@ type Props = {
 };
 
 // The backend converts all token IDs to hexadecimals; here, we convert back
-// TODO: Fix parseInt() for long strings (> 21 characters?), when JS converts to scientific https://gallery.so/robin/21Qt76R6bIyM7fIDsBsgU9DrKCg/21R0nPXR1nlqXP9UD8hn7CQUsUP
-const hexHandler = (str: string) => parseInt(str, 16);
+const hexHandler = (str: string) => {
+  if (str.length % 2) {
+    str = '0' + str;
+  }
```

Comment:
what exactly is the purpose of this?  if the string length is even we add a `0` in front? i think we should add a comment to explain!

Reply:
would love a comment here!